utils/merkletrie/fsnoder/file.go

```diff
@@ -0,0 +1,73 @@
+package fsnoder
+
+import (
+	"bytes"
+	"fmt"
+	"hash/fnv"
+
+	"gopkg.in/src-d/go-git.v4/utils/merkletrie/noder"
```

Comment:
It is needed though, to avoid a circular dependency **in the future**:

- `merkletrie` will import `merkletrie/frame` because the difftree function uses frames.
- `merkletrie/frames` will import `merkletrie` because frames stores noders.

The only way to solve this is to extract the noder to its own package, sadly.